keyboards/planck/keymaps/vuldriel/keymap.c

```diff
@@ -0,0 +1,870 @@
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |   S  |   K  |   W  |   R  |   *  |   *  |   R  |   B  |   G  |   S  |   Z  |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Exit |      |      |   A  |   O  |             |   E  |   U  |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_PLOVER] = LAYOUT_planck_grid(
+    KC_1   , KC_1   , KC_1   , KC_1, KC_1, KC_1   , KC_1   , KC_1, KC_1, KC_1   , KC_1   , KC_1   ,
+    XXXXXXX, KC_Q   , KC_W   , KC_E, KC_R, KC_T   , KC_Y   , KC_U, KC_I, KC_O   , KC_P   , KC_LBRC,
+    XXXXXXX, KC_A   , KC_S   , KC_D, KC_F, KC_G   , KC_H   , KC_J, KC_K, KC_L   , KC_SCLN, KC_QUOT,
+    EXT_PLV, XXXXXXX, XXXXXXX, KC_C, KC_V, XXXXXXX, XXXXXXX, KC_N, KC_M, XXXXXXX, XXXXXXX, XXXXXXX
+),
+
+/* Adjust (Lower + Raise)
+ *               v------------------------RGB CONTROL--------------------v
+ * ,-----------------------------------------------------------------------------------.
+ * |Reset | Debug| Hue+ | Hue- | Sat+ | Sat- | Bri+ | Bri- |Mode +|RGBTog|Plover| Bksp |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |BL Bth|MouseL|MouseD|MouseU|MouseR|Dacro1|Dacro2|Mouse1|Mouse2|Mouse3|Mouse4| LOCK |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |BL Cyc|Beeps-|Beeps+|Mus on|Musoff|Rec 1 |Rec 2 |MWhelL|MWhelD|MWhelU|MWhelR|Sleep |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |BL Tog|MUSmod|AudTog|Clicky|      |             |      |GIRAFF|Macro2|Macro3|Macro4|
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_planck_grid(
+    RESET  , DEBUG  , RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD        , RGB_VAI        , RGB_VAD, RGB_MODE_FORWARD, RGB_TOG, PLOVER , KC_BSPC,
+    BL_BRTG, KC_MS_L, KC_MS_D, KC_MS_U, KC_MS_R, DYN_MACRO_PLAY1, DYN_MACRO_PLAY2, KC_BTN1, KC_BTN2         , KC_BTN3, KC_BTN4, KC_LOCK,
+    BL_STEP, CK_DOWN, CK_UP  , MU_ON  , MU_OFF , DYN_REC_START1 , DYN_REC_START2 , KC_WH_L, KC_WH_D         , KC_WH_U, KC_WH_R, KC_SLEP,
+    BL_TOGG, MU_MOD , AU_TOG , CK_TOGG, _______, _______        , _______        , _______, GIRAFFACT       , XXXXXXX, XXXXXXX, XXXXXXX
+)
+
+};
+
+bool is_alt_tab_active = false;
+uint16_t alt_tab_timer = 0;
+
+int cur_dance (qk_tap_dance_state_t *state) {
+  if (state->count == 1) {
+    if (state->interrupted || !state->pressed)  return SINGLE_TAP;
+    //key has not been interrupted, but they key is still held. Means you want to send a 'HOLD'.
+    else return SINGLE_HOLD;
+  } 
+  else return 8; //magic number. At some point this method will expand to work for more presses
+}
+
+//instanalize an instance of 'tap' for the 'x' tap dance.
+static tap xtap_state = {
```

Comment:
Just a heads up here, you explicitly want one per dance.  Otherwise, if you hit another tap dance key while the current one is "active", it will get stuck, since it's sharing the tap state between all of them.

Reply:
Thanks so much for this feedback! I'm working on it right now for it to make more sense and be a lot simpler.